floodwatch/src/js/components/Compare.js

```diff
@@ -0,0 +1,153 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Row, Col } from 'react-bootstrap';
+import $ from 'jquery';
+import {FilterParent} from './FilterParent';
+import AdBreakdown from '../../stubbed_data/adbreakdown_response.json';
+import Filters from '../../stubbed_data/filter_response.json';
+
+// import '../../Compare.css';
+
+type CompareState = {}
+export class Compare extends Component {
+  state: CompareState;
+
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <Row>
+        <Col xs={12}>
+        <CompareContainer/>
+        </Col>
+      </Row>
+    )
+  }
+}
+
+type CompareContainerState = {
+  leftOptions: JSON;
+  rightOptions: JSON;
+  currentTopic: string;
+}
+
+function CompareContainerInitialState() {
+  return {
+    leftOptions: Filters.presets[0].filters,
+    rightOptions: Filters.presets[1].filters,
+    leftData: {},
+    rightData: {},
+    currentTopic: 'Food'
+  }
+}
+
+export class CompareContainer extends Component {
+  state: CompareContainerState;
+
+  constructor(props) {
+    super(props);
+    this.state = CompareContainerInitialState()
+  }
+
+  componentDidMount() {
+    this.setState({
+      leftData: AdBreakdown.filterA,
+      rightData: AdBreakdown.filterB
+    })
+  }
+
+  createSentence(options) {
+    let sentence = 'Floodwatch users';
+    if (options.length == 0) {
+      sentence = 'All ' + sentence;
+      return sentence
+    }
+
+    if (options[0].name == 'data') {
+      return 'You'
+    }
+
+    options.map((opt, key) => {
+      let logic = ''
+      let choices = ''
+      if (opt.logic == 'NOT') { 
+        logic = ' Non-';
+        choices = opt.choices.join(', non-')
+      } else {
+        choices = opt.choices.join(', ')
+      }
+      sentence = logic + choices + ' ' + sentence
+    })
+
+    return sentence
+  }
+
+  updateMouseOver(newTopic) {
```

Comment:
If you type this function here, you can enforce the type down the chain, so this would be `updateMouseOver(newTopic: Topic): void` (see comment on FilterParent.js#13)